PyMPDATA/solver.py

```diff
@@ -36,9 +36,15 @@ def call(self, flux, g_factor, step, iteration):  # pylint: disable=unused-argum
 
 
 class Solver:
-    """solution orchestrator requireing prior instantiation of: a `Stepper`,
-    a scalar advectee field, a vector advector field and optionally
-    a scala g_factor field"""
+    """Solution orchestrator requiring prior instantiation of: a `Stepper`,
+    a scalar advectee field (that which is advected),
+    a vector advector field (that which advects),
+    and optionally a scalar g_factor field (used in some cases of the advection equation).
+    Note: in some cases of advection, i.e. momentum advection,
+    the advectee can act upon the advector.
+    See [Dave Randall's AT604 notes]
+    (https://hogback.atmos.colostate.edu/group/dave/at604pdf/AT604_LaTeX_Book.pdf)
```

Comment:
this one is a 500+ pages book - I doubt the link is helpful :) 
perhaps, we could point to the PyMPDATA example where it is actually exemplified (i.e., shallo water)?